erpnext/crm/doctype/opportunity/opportunity.py

```diff
@@ -490,20 +490,21 @@ def get_customer_from_opportunity(source):
 
 
 @frappe.whitelist()
-def submit_communication(name, contact_date, remarks):
+def submit_communication(name, contact_date, remarks, submit_follow_up=False):
 	if not remarks:
-		frappe.throw(_('Remarks are mandatory for follow up'))
+		frappe.throw(_('Remarks are mandatory for Communication'))
+
+	if not frappe.db.exists('Opportunity', name):
+		frappe.throw(_("Opportunity does not exist"))
 
 	opp = frappe.get_cached_doc('Opportunity', name)
-	follow_up = [f for f in opp.contact_schedule if not f.contact_date]
-	if follow_up:
-		follow_up[0].contact_date = getdate(contact_date)
-	else:
-		opp.append("contact_schedule", {
-			"contact_date": contact_date,
-		})
 
-	opp.save()
+	if submit_follow_up:
```

Comment:
do `if cint(submit_follow_up)` .. since frappe api sometimes passes boolean or integer as string and make sure you send integer through API instead of "true" or "false"

Reply:
alright